camunda-formio-tasklist-vue/src/components/TaskList.vue

```diff
@@ -798,28 +771,77 @@ export default class Tasklist extends Mixins(TaskListMixin) {
     max: number
   ) {
     this.selectedfilterId = filterId;
-    await CamundaRest.filterTaskListPagination(
+    const paginatedTaskResults = await CamundaRest.filterTaskListPagination(
       this.token,
       filterId,
       requestData,
       first,
       max,
       this.bpmApiUrl
-    ).then((result) => {
-      this.tasks = result.data;
-    });
+    );
+    this.tasks = paginatedTaskResults.data;
+  }
+
+  async onUserSearch (search: string, loading: any) {
+    if(search.length) {
+      loading(true);
+      this.reviewerUsersList = [];
+    }
+
+    if(this.selectSearchType === "firstName") {
+      const firstNameUserList = await CamundaRest.getUsersByFirstNameGroups(this.token, this.bpmApiUrl, search, reviewerGroup);
+      if(firstNameUserList) {
+        this.reviewerUsersList = [];
+        firstNameUserList.data.forEach((user: UserPayload) => {
+          this.reviewerUsersList.push({
+            code: user.id,
+            email: user.email!,
+            firstName: `${user.firstName!} ${user.lastName!}`,
+            lastName: `${user.lastName!} ${user.firstName!}`,
+          });
+        });
+        loading(false);
+      }
+    }
+
+    if(this.selectSearchType === "lastName") {
+      const lastNameUserList = await CamundaRest.getUsersByLastNameGroups(this.token, this.bpmApiUrl, search, reviewerGroup);
+      if(lastNameUserList) {
+        this.reviewerUsersList = [];
+        lastNameUserList.data.forEach((user: UserPayload) => {
+          this.reviewerUsersList.push({
+            code: user.id,
+            email: user.email!,
+            firstName: `${user.firstName!} ${user.lastName!}`,
+            lastName: `${user.lastName!} ${user.firstName!}`,
+          });
+        });
+        loading(false);
+      }
+    }
+
+    if(this.selectSearchType === "email") {
+      const emailUserList = await CamundaRest.getUsersByEmailGroups(this.token, this.bpmApiUrl, search, reviewerGroup);
+      if(emailUserList) {
+        this.reviewerUsersList = [];
+        emailUserList.data.forEach((user: UserPayload) => {
+          this.reviewerUsersList.push({
+            code: user.id,
+            email: user.email!,
+            firstName: `${user.firstName!} ${user.lastName!}`,
+            lastName: `${user.lastName!} ${user.firstName!}`,
+          });
+        });
+        loading(false);
```

Comment:
what happens to loading if it does not falls into any of the above 3 conditions??

Reply:
![image](https://user-images.githubusercontent.com/70306694/134508279-b305234e-bde5-44af-98fe-22536f545e9b.png)

It will be be in loading as true state if it's not in any of the three variables ie `firstName`, `lastName`, `email`.